src/System.Management.Automation/engine/hostifaces/PowerShell.cs

```diff
@@ -860,6 +860,20 @@ public static PowerShell Create(InitialSessionState initialSessionState)
             return result;
         }
 
+        /// <summary>
+        /// Constructs an empty PowerShell instance; a script or command must be added before invoking this instance.
+        /// </summary>
+        /// <param name="runspace">Runspace in which to invoke commands.</param>
+        /// <returns>An instance of PowerShell.</returns>
+        public static PowerShell Create(Runspace runspace)
+        {
+            PowerShell result = Create();
+
+            result.Runspace = runspace;
```

Comment:
Should we consider checking runspace state? Also, to match the other overloads behavior should the runspace be opened is it is in 'BeforeOpening' state?

Reply:
Runspace state checks are already built-in when you invoke PowerShell.Invoke. Plus, there is no need to check state here. You can create the PowerShell instance regardless of the runspace state.

For the second question, this overload of the Create method takes a runspace that you have created and are managing elsewhere. In the other overload that you are referring to (the one in InitialSessionState), that overload creates the runspace that it assigns to the PowerShell instance, and since it is being done from the InitialSessionState that you want to load, it makes sense to open the runspace because that is the only way to load that session state into the runspace. In this case, opening the runspace is a task that is done elsewhere so it would be wrong to call open again.

Reply:
I think it makes sense to not check for Runspace state and perform actions like calling `Open` when it's `BeforeOpening`, because it's possible the Runspace is being opened in another thread while we are creating a `PowerShell` instance using this method. Let's just be dumb and behave more predictably.

However, I think adding a comment here to explain why we don't check for state or so would be very useful.